### PR TITLE
rename voice_assistant.state to agent.state

### PR DESCRIPTION
**Pull request by bcherry**
Following conversation in Slack.

See https://github.com/livekit/components-js/pull/976 as well

### CODE REVIEW

File: livekit-agents/livekit/agents/proto.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from typing import Literal, Union
+
+AgentState = Union[Literal["initializing", "listening", "thinking", "speaking"], str]
```
[the remark this answers is not in the document]
Comment (theomonnom) [reply livekit-agents/livekit/agents/proto.py:3]: kk make sense, let's make sure we never expose the file itself, only the content

**Review by theomonnom — APPROVED** [review]

File: livekit-agents/livekit/agents/proto.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bcherry) [reply livekit-agents/livekit/agents/proto.py:5]: yeah i think that's what's done in the main `__init__.py`? also I just updated it to put the attr name into a constant too, lmk if i did this in the most pythonic way or not!

**Comment by lukasIO** [comment]
Are we expecting users to add custom states directly to the `VoiceAssistant` class ? 
If we do, we should make `_update_state` a public method. 
If not, it might make sense to keep `VoiceAssistantState` as a narrowed down type of `AssistantState` where assistant state could consist only of either `str` or some more generally applicable states than `listening` and `speaking` - which sound very much voice-only related.

**Comment by bcherry** [comment]
> Are we expecting users to add custom states directly to the `VoiceAssistant` class ? If we do, we should make `_update_state` a public method. If not, it might make sense to keep `VoiceAssistantState` as a narrowed down type of `AssistantState` where assistant state could consist only of either `str` or some more generally applicable states than `listening` and `speaking` - which sound very much voice-only related.

yeah that could make sense - i think that would be easy to change later as we think through it so i'm going to merge this once the tests pass

File: .changeset/curly-rats-burn.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"livekit-agents": patch
```
Comment (nbsp) [inline .changeset/curly-rats-burn.md:2]:
```suggestion
"livekit-agents": minor
```

**Review by nbsp — CHANGES_REQUESTED** [review]

**Comment by lukasIO** [comment]
>  i think that would be easy to change later as we think through it

if we choose to narrow down a type lateron, I think that will be a breaking change?

**Comment by bcherry** [comment]
> > i think that would be easy to change later as we think through it
> 
> if we choose to narrow down a type lateron, I think that will be a breaking change?

in this version, VoiceAssistantState (as a type) doesn't exist at all in the public parts of the framework - it's all internal.

**Comment by bcherry** [comment]
@theomonnom and I discussed offline and decided it would be a good idea to use a `lk.` prefix on attributes we define, to avoid collisions with user-defined attributes.

**Comment by lukasIO** [comment]
> in this version, VoiceAssistantState (as a type) doesn't exist at all in the public parts of the framework - it's all internal.

makes sense, but we do expose the changed name in https://github.com/livekit/components-js/pull/976

**Comment by bcherry** [comment]
> makes sense, but we do expose it in [livekit/components-js#976](https://github.com/livekit/components-js/pull/976)

yeah the values are exposed but not the constrained python type. but maybe i misunderstood your suggestion - were you thinking it's good to keep the `voice_assistant.state` attribute too? in addition to a `agent.state` attribute that has less granularity?

**Pull request merged**